src/click/decorators.py

```diff
@@ -153,11 +153,29 @@ def _make_command(
     )
 
 
+@t.overload
```

Comment:
Add `import typing` and use `@typing.overload` for this specific case, I've been doing the same elsewhere around the pallets code.

Reply:
Done in newest commit